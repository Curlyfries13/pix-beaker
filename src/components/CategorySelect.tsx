import React, { useState } from 'react';
import { categories } from 'categories';

const CategorySelect = ({ category }: Props) => {
  const [selected, setSelected] = useState(category === undefined ? '' : category)
  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    setSelected(event.target.value);
  }
  return (
    <select
      value={selected}
      onChange={handleChange}>
      {Object.keys(categories).map((key: string) => {
        return (
          <option key={key} value={categories[key].display}>
            {categories[key].display}
          </option>
        );
      })}
    </select>
  )
}
type Props = { category?: string; };

export default CategorySelect;
