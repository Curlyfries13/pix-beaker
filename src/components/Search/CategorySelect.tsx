import React, { useState } from 'react';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import { withStyles } from '@material-ui/core/styles'

import { categories } from 'categories';

// This component is the customized Selection component for selecitng categoires
// from the Pixabay API.
const CategorySelect = ({ category, onSelect }: Props) => {
  const [selected, setSelected] = useState(category === undefined ? '' : category)
  const handleChange = (event: any) => {
    setSelected(event.target.value);
    onSelect(event.target.value);
  }

  const CategorySelect = withStyles({
    root: {
      'min-width': '85px'
    }
  })(Select)

  return (
    <FormControl>
      <InputLabel id="category-select">Category</InputLabel>
      <CategorySelect
        labelId="category-select"
        value={selected}
        onChange={handleChange}>
        {/* TODO: add a NONE selection */}
        {Object.keys(categories).map((key: string) => {
          return (
            <MenuItem key={key} value={key}>
              {categories[key].display}
            </MenuItem>
          );
        })}
      </CategorySelect>
    </FormControl>
  )
}
type Props = { category?: string; onSelect: Function };

export default CategorySelect;
