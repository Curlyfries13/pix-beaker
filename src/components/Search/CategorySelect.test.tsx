import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import CategorySelect from './CategorySelect';

it('matches snapshot', async () => {
  const component: RenderResult = render(<CategorySelect />)
  expect(component).toMatchSnapshot();
})
