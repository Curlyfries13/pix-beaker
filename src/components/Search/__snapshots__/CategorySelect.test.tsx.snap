// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          id="category-select"
        >
          Category
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="category-select"
            class="MuiSelect-root WithStyles(ForwardRef(Select))-root-1 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            role="button"
            tabindex="0"
          >
            <span>
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        id="category-select"
      >
        Category
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="category-select"
          class="MuiSelect-root WithStyles(ForwardRef(Select))-root-1 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          role="button"
          tabindex="0"
        >
          <span>
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
